dataverse/webapi/C#-NETx/BulkOperations/Utility.cs

```diff
@@ -272,8 +288,186 @@ public static async Task DeleteExampleTable(Service service, string tableSchemaN
+
+                            return asyncJobResponse["Message"].ToString();
+                        }
+                        timeOutInterval -= pollingInterval;
+                    }
+                }
+                else
+                {
+                    waitForKeyStatus = true;
+                }
+                return "Pending";
+            }
+            catch (Exception ex)
+            {
+                return ex.Message;
+            }
+        }
+
+        /// <summary>
+		/// Returns the entity records
+		/// </summary>
+		/// <example>accounts</example>
+		/// <example>tasks(6b5941dd-d175-e511-80d4-00155d2a68d1)</example>
+		/// <param name="query">The relative URL of the message request</param>
+		/// <param name="httpRequestHeaders">Http request header details</param>
+		/// <returns>Returns the response object</returns>
+		public static async Task<JObject> GetEntityRecords(
+            Service service, 
+            string query, 
+            Dictionary<string, string> httpRequestHeaders = null)
+        {
+            var request = new RetrieveMultipleRequest(query);
+
+            // TODO: You can use service.SendAsync<RetrieveMultipleResponse>(request) to get 
+            // a typed RetrieveMultipleResponse object returned with a Records property that could
+            //  replace this whole method.
+            var response = await service.SendAsync(request).ConfigureAwait(false);
+
+            JObject jsonResponse = JsonConvert.DeserializeObject<JObject>(await response.Content.ReadAsStringAsync().ConfigureAwait(false));
+            return jsonResponse;
+        }
+
+        /// <summary>
+		/// Get the response url from Response Message
+		/// </summary>
+		/// <param name="response"></param>
+		/// <returns></returns>
+		private static string GetUrlFromResponse(HttpResponseMessage response)
+        {
+            var responseUrl = response.Headers.GetValues("OData-EntityId").FirstOrDefault().ToString();
+
+            var returnUrl = responseUrl.Substring(responseUrl.IndexOf("EntityDefinitions"));
+
+            return returnUrl;
+        }
+
+        /// <summary>
+		/// Creates alternate key to the entity requested.
+		/// </summary>
+		/// <param name="entitySchemaNane">Entity schema name</param>
+		/// <param name="schemaName">Schema name.</param>
+		/// <param name="displayName">Display name.</param>
+		/// <param name="keyAttributes">List of Key Attributes.
+		/// These attributes should exists in the specified entity.
+		/// These attributes must be of type Decimal or Integer or String.</param>
+		/// <returns>Return True if success</returns>
+		private static async Task<HttpResponseMessage> CreateAlternateKeyToEntityAsync(Service service, 
+            string entitySchemaName, 
+            string schemaName, 
+            string displayName, 
+            List<string> keyAttributes, 
+            bool isSynchronous = false,  //TODO: Remove unused parameters
```

Comment:
I don't think you need this message, but you should remove any unused parameters.